parsers/saarland.py

```diff
@@ -0,0 +1,199 @@
+from urllib.request import *
+
+import json
+import datetime
+
+from pyopenmensa.feed import LazyBuilder
+
+from utils import Parser, Source
+
+
+API_VERSION = "1"
+# please don't copy this key, contact felix@fefrei.de to get your own instead
+API_KEY = "A1RT5M56" + "ibsGaGNZBJQg"
+APP_VERSION = "1"
+SUPPORTED_LANGUAGES = ["de", "en"]
+
+URL_BASE = (
+    'https://mensaar.de/api/%s/%s/%s/%s/'
+    % (API_VERSION, API_KEY, APP_VERSION, SUPPORTED_LANGUAGES[0])
+)
+URL_BASE_DATA = "getBaseData"
+URL_MENU = "getMenu/"
+
+UTC_DATE_STRING = "%Y-%m-%dT%H:%M:%S.%fZ"
+
+ROLES = {
+    's': 'student',
+    'g': 'other',
+    'm': 'employee'
+}
+
+SPECIAL_NOTICES = [
+    'su',   # geschwefelt
+    'bl',   # geschwärzt
+    'gw',   # gewachst
+    've',   # vegetarisch
+    'bio',  # biologisches Essen
+    'nla',  # laktosefrei
+    'vn'    # vegan
+]
+
+
+def get_notices(notices, sub_notices=False):
+    global base_data
+
+    notices_special = []
+    notices_allergens = []
+    notices_others = []
+    notices_others_last = []
+
+    for notice_short in notices:
+        notice = base_data['notices'][notice_short]
+        if notice['isAllergen']:
+            if sub_notices:
+                notices_allergens.append(notice['displayName'])
+            else:
+                notices_allergens.append(
+                    'enthält %s' % notice['displayName'])
+        else:
+            if notice_short == 'nsf':  # ohne Schweinefleisch
```

Comment:
Another special case is `nsf`, the only value that states explicitly the _absence_ of an ingredient, so here too, I had to avoid a `note` as `mit ohne Schweinefleisch`.

Reply:
Indeed, I'd say this is an oversight in the API. It's fixed now:

```
        "nu": {
            "displayName": "Schalenfrüchte (Nüsse)",
            "isAllergen": true,
            "isNegated": false
        },
        "nsf": {
            "displayName": "ohne Schweinefleisch",
            "isAllergen": false,
            "isNegated": true
        },
```

I hope this helps!

Reply:
Wow, you're quick! Thanks a lot!

Reply:
That's what happens when the API isn't operated by the Studentenwerk itself, but an enthusiastic student – API changes are just one push away 🎉

Reply:
Could you incorporate this change?

Reply:
Done.